contracts/Market.sol

```diff
@@ -102,6 +103,7 @@ contract Market is
    */
   struct FulfillOrderData {
     bool chargeFee;
+    uint256 feePercentage;
```

Comment:
we can't because `chargeFee` is used to predicate logic in the order fulfillment, but we often have scenarios (cash and special orders) where we need to specify what fee percentage was used off-chain (like the custom fee option we're introducing here) but _don't actually charge that fee during the on-chain purchase_ 

so we often have a scenario where `chargeFee` is false but the fee we're reporting in the emitted event is non-zero.  kinda weird, but this is how patrick and business folks want it so that we can have accurate on-chain accounting for fees that were collected